PWGCF/MultiparticleCorrelations/Tasks/multiparticle-correlations-ar.cxx

```diff
@@ -71,314 +108,366 @@ static constexpr std::string_view TrackVariableNames[kLAST_TrackVariable] = {"Pt
+  bool SurviveTrackCuts(TrackObject track)
   {
-    registry.fill(HIST(AR::TCrecoBefore) + HIST(AR::TrackVariableNames[AR::kPT]), track.pt());
-    registry.fill(HIST(AR::TCrecoBefore) + HIST(AR::TrackVariableNames[AR::kPHI]), track.phi());
-    registry.fill(HIST(AR::TCrecoBefore) + HIST(AR::TrackVariableNames[AR::kETA]), track.eta());
-    registry.fill(HIST(AR::TCrecoBefore) + HIST(AR::TrackVariableNames[AR::kCHARGE]), track.sign());
-    registry.fill(HIST(AR::TCrecoBefore) + HIST(AR::TrackVariableNames[AR::kDCAZ]), track.dcaZ());
-    registry.fill(HIST(AR::TCrecoBefore) + HIST(AR::TrackVariableNames[AR::kDCAXY]), track.dcaXY());
-    registry.fill(HIST(AR::TCrecoBefore) + HIST(AR::TrackVariableNames[AR::kTPCCLUSTERS]), track.tpcNClsFound());
-    registry.fill(HIST(AR::TCrecoBefore) + HIST(AR::TrackVariableNames[AR::kTPCCROSSEDROWS]), track.tpcNClsCrossedRows());
-    registry.fill(HIST(AR::TCrecoBefore) + HIST(AR::TrackVariableNames[AR::kTPCCHI2]), track.tpcChi2NCl());
-    registry.fill(HIST(AR::TCrecoBefore) + HIST(AR::TrackVariableNames[AR::kITSCLUSTERS]), track.itsNCls());
-  }
-  template <typename TrackInstance>
-  void FillTrackControlHistAC(TrackInstance const& track, HistogramRegistry& registry)
-  {
-    registry.fill(HIST(AR::TCrecoAfter) + HIST(AR::TrackVariableNames[AR::kPT]), track.pt());
-    registry.fill(HIST(AR::TCrecoAfter) + HIST(AR::TrackVariableNames[AR::kPHI]), track.phi());
-    registry.fill(HIST(AR::TCrecoAfter) + HIST(AR::TrackVariableNames[AR::kETA]), track.eta());
-    registry.fill(HIST(AR::TCrecoAfter) + HIST(AR::TrackVariableNames[AR::kCHARGE]), track.sign());
-    registry.fill(HIST(AR::TCrecoAfter) + HIST(AR::TrackVariableNames[AR::kDCAZ]), track.dcaZ());
-    registry.fill(HIST(AR::TCrecoAfter) + HIST(AR::TrackVariableNames[AR::kDCAXY]), track.dcaXY());
-    registry.fill(HIST(AR::TCrecoAfter) + HIST(AR::TrackVariableNames[AR::kTPCCLUSTERS]), track.tpcNClsFound());
-    registry.fill(HIST(AR::TCrecoAfter) + HIST(AR::TrackVariableNames[AR::kTPCCROSSEDROWS]), track.tpcNClsCrossedRows());
-    registry.fill(HIST(AR::TCrecoAfter) + HIST(AR::TrackVariableNames[AR::kTPCCHI2]), track.tpcChi2NCl());
-    registry.fill(HIST(AR::TCrecoAfter) + HIST(AR::TrackVariableNames[AR::kITSCLUSTERS]), track.itsNCls());
+    bool flag = true;
+
+    if (!AR::SurviveCut(cfgPT.value, track.pt()) ||
+        !AR::SurviveCut(cfgPHI.value, track.phi()) ||
+        !AR::SurviveCut(cfgETA.value, track.eta()) ||
+        !AR::SurviveCut(cfgCHARGE.value, track.sign()) ||
+        !AR::SurviveCut(cfgDCAZ.value, track.dcaZ()) ||
+        !AR::SurviveCut(cfgDCAXY.value, track.dcaXY()) ||
+        !AR::SurviveCut(cfgTPCCLUSTERS.value, track.tpcNClsFound()) ||
+        !AR::SurviveCut(cfgTPCCROSSEDROWS.value, track.tpcNClsCrossedRows()) ||
+        !AR::SurviveCut(cfgTPCCHI2.value, track.tpcChi2NCl()) ||
+        !AR::SurviveCut(cfgITSCLUSTERS.value, track.itsNCls())) {
+      flag = false;
+    }
+
+    return flag;
```

Comment:
You can just return AND of meeting all conditions: `return AR::SurviveCut(...) && AR::SurviveCut(...) && ...`

Reply:
Fixed.